web_src/js/features/repo-issue-pr-status.js

```diff
@@ -2,11 +2,15 @@ export function initRepoPullRequestCommitStatus() {
   for (const btn of document.querySelectorAll('.commit-status-hide-checks')) {
     const panel = btn.closest('.commit-status-panel');
     const list = panel.querySelector('.commit-status-list');
+    let hasScrollBar = null;
     btn.addEventListener('click', () => {
+      if (hasScrollBar === null && !list.style.overflow) {
+        hasScrollBar = list.scrollHeight > list.clientHeight;
+      }
+      list.style.overflow = hasScrollBar ? '' : 'hidden'; // hide the flickering scrollbar when hiding if there was no scrollbar
```

Comment:
```suggestion
    let hasScrollBar = false;
    btn.addEventListener('click', () => {
      if (!hasScrollBar && !list.style.overflow) {
        hasScrollBar = list.scrollHeight > list.clientHeight;
      }
      list.style.overflow = hasScrollBar ? '' : 'hidden'; // hide the flickering scrollbar when hiding if there was no scrollbar
```

Prefer boolean.

Reply:
Commited this in https://github.com/go-gitea/gitea/pull/27932/commits/ae0125d9660b29f4467fd8a7c73c14724cd2b6f5.